### PR TITLE
io.github.hmlendea.geforcenow-electron

### PR DESCRIPTION
**Pull request by roypen**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **https://github.com/hmlendea/geforcenow-electron**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build Io.GitHub.electron.geforcenow

**Comment by nedrichards** [comment]
@roypen if this is using a github app-id you can tick the box in the description.

File: io.github.electron.geforcenow.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+app-id: io.github.electron.geforcenow
```
Comment (barthalion) [inline io.github.electron.geforcenow.yaml:1]: The proper ID seems to be `io.github.hmlendea.geforcenow-electron`.

File: io.github.electron.geforcenow.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+base-version: '20.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node14
+separate-locales: false
+command: start-geforcenow-electron
+
+finish-args:
+  - '--share=ipc'
+  - '--socket=x11'
+  - '--socket=wayland'
```
Comment (barthalion) [inline io.github.electron.geforcenow.yaml:16]: Does it actually work under Wayland or fallbacks to xwayland?

File: io.github.electron.geforcenow.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  <name>GeForce Now Electron</name>
+  <project_license>CC0-1.0</project_license>
+  <developer_name>hmlendea</developer_name>
+  <summary>Cloud gaming electron app</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <url type="homepage">https://github.com/hmlendea/geforcenow-electron</url>
+  <url type="help">https://github.com/hmlendea/geforcenow-electron/issues</url>
+  <url type="bugtracker">https://github.com/hmlendea/geforcenow-electron/issues</url>
+  <description>
+    <p>GeForce Now application written in Electron, wrapping around the GFN web browser implementation.</p>
```
Comment (barthalion) [inline io.github.electron.geforcenow.metainfo.xml:14]: I think it should be stated it's not an official app and it's not affiliated with Nvidia. See DataGrip for an example: https://flathub.org/apps/details/com.jetbrains.DataGrip

File: io.github.electron.geforcenow.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline io.github.electron.geforcenow.metainfo.xml:1]: Ideally metainfo, screenshots, desktop file and icon should be maintained in the upstream repo and not here.

**Comment by barthalion** [comment]
bot, build io.github.electron.geforcenow

File: io.github.electron.geforcenow.metainfo.xml (hunk 3 above)
Comment (roypen) [reply io.github.electron.geforcenow.metainfo.xml]: I've changed the description

File: io.github.electron.geforcenow.yaml (hunk 2 above)
Comment (roypen) [reply io.github.electron.geforcenow.yaml]: Yes, the application works under wayland.

**Comment by AsciiWolf** [comment]
The AppStream metadata file needs to be correctly installed. Add this to build-commands:
```
install -Dm 644 io.github.hmlendea.geforcenow-electron.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
```

**Comment by roypen** [comment]
> The AppStream metadata file needs to be correctly installed. Add this to build-commands:
> 
> ```
> install -Dm 644 io.github.hmlendea.geforcenow-electron.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
> ```

OMG, I overlooked that. Thanks

**Comment by AsciiWolf** [comment]
bot, build io.github.electron.geforcenow

**Comment by AsciiWolf** [comment]
bot, build io.github.hmlendea.geforcenow-electron

**Comment by AsciiWolf** [comment]
bot, build io.github.hmlendea.geforcenow-electron

**Comment by roypen** [comment]
> Build [38342 failed](https://flathub.org/builds/#/builders/32/builds/38342)

Maybe it's because the upstream repo contains a desktop file which points to a different icon filename. Does the upstream desktop file have priority over this PR?

**Comment by AsciiWolf** [comment]
Only if it would have the same name. But if there are upstream desktop/icon files being installed, you should use them, just [add](https://docs.flatpak.org/en/latest/manifests.html#file-renaming) `rename-desktop-file` and `rename-icon` to the manifest.

**Comment by roypen** [comment]
> Only if it would have the same name. But if there are upstream desktop/icon files being installed, you should use them, just [add](https://docs.flatpak.org/en/latest/manifests.html#file-renaming) `rename-desktop-file` and `rename-icon` to the manifest.

They have. I'm not a maintainer of the upstream repo, so I have to wait until PR is reviewed. For now, I will change desktop and icon file to match app id.
Sorry for so many commits, but It is my first flatpak and these problems didn't appear when I build locally.

**Comment by AsciiWolf** [comment]
bot, build io.github.hmlendea.geforcenow-electron

**Comment by barthalion** [comment]
bot, build io.github.hmlendea.geforcenow-electron

**Comment by barthalion** [comment]
bot, build io.github.hmlendea.geforcenow-electron

**Pull request closed without merging**